code/drasil-build/lib/Build/Drasil/Make/Print.hs

```diff
@@ -25,8 +25,8 @@ printRule (R t d _ c) = printTarget t d $+$ printCmds c
 
 -- | Gathers all rules to abstract targets and tags them as phony.
 printPhony :: [Rule] -> Doc
-printPhony = (<+>) (text ".PHONY:") . hsep . tail . map (\(R t _ _ _) -> text $ renderMS t) . 
-  filter (\(R _ _ t _) -> t == Abstract)
+printPhony = (<+>) (text ".PHONY:") . hsep . noFirstElement . tail . map (\(R t _ _ _) -> text $
```

Comment:
The reason that there are two tail calls here is because otherwise, the last line of a makefile would read `PHONY: # Generated by Drasil Version build run doc` instead of `PHONY: build run doc`, which would make the remaining part of the line a comment

Reply:
Oh, that sounds odd. The comment in the Makefile shouldn't be a `Rule` type, it should just be a comment. Looking at Drasils [definition of a `Makefile`](https://github.com/JacquesCarette/Drasil/blob/0241639a27891e96ebe1c22733a833a95ea1652f/code/drasil-build/lib/Build/Drasil/Make/AST.hs#L7-L11), it looks like it isn't quite sufficient. A `Makefile` isn't exactly a sequence of rule definitions, which is why we (well, you) had to use `Rule` to fix that. It would be good to have a base PR (for this PR) that adds just enough structure to the definition of `Makefile`s such that we can have comments interspersed between rule definitions. Afterwards, we can improve the definition of `Makefile`s to use the same 'statements' system that `Makefile`s use.

What do you think?

Reply:
I agree! The `Makefile` should be able to have comments separate from the Rules. I like the idea of creating another PR  which provides a further structure to the way `Makefile`  as that would improve the creation of `Makefile`s.